examples/main/main.cpp

```diff
@@ -236,6 +236,11 @@ int main(int argc, char ** argv) {
     }
 
     if (params.interactive) {
+        if (params.prompt_cache_all) {
+            fprintf(stderr, "error: --prompt-cache-all not supported in interactive mode yet\n");
+            return 1;
+        }
+
```

Comment:
This check should be done before we load the model. Move this to `gpt_params_parse` in `common.cpp`, right before the `invalid_param` check on line [344](https://github.com/ggerganov/llama.cpp/blob/a7298bd178828a4f7b798b4e256ac413662ed890/examples/common.cpp#L344-L348) and instead of returning `1`, set `invalid_param = true`.

Reply:
Sure, I'll do that. I admit I was taking a shortcut here to catch all the params/code paths that set interactive (e.g., instruct). I'd probably opt for a custom message still though because the `invalid_param` case doesn't seem to be a great fit:
```
error: invalid parameter for argument: %s
```

Reply:
Oh, I see. Yeah, you'd have to check `instruct`, `interactive_first`, and `antiprompt.size()`. Those three cover all the cases though?

And yeah, I think it's still a good idea to print out the same message you had above. It's probably good enough on it's own but setting `invalid_params` to true will show them the rest of the notes about the options listed in the help.

`error: invalid parameter for argument: %s` only tells them the name of `main` before listing out all the options.